File: lowfare-flights-search-service/src/main/java/hr/msutic/lowfareflights/flightoffer/SearchCommand.java
package hr.msutic.lowfareflights.flightoffer;

import lombok.Value;

import java.time.LocalDate;

@Value
public class SearchCommand {

  private final String origin;
  private final String destination;
  private final LocalDate departureDate;
  private final LocalDate returnDate;
  private final int adults;
  private final String currency;

}

File: lowfare-flights-search-client/src/app/services/search-command.ts
export class SearchCommand {

  departureAirport: string;
  destinationAirport: string;
  departureDate: string;
  returnDate: string;
  passengers: number;
  currency: string;

  public toString = (): string => {
    return `SearchCommand (
    departureAirport: ${this.departureAirport},
    departureDate: ${this.departureDate},
    returnDate: ${this.returnDate},
    passengers: ${this.passengers},
    currency: ${this.currency},
    )`;
  }
}

